lib/ansible/constants.py

```diff
@@ -333,6 +333,7 @@ def load_config_file():
 # that it may be a security risk to do so.
 ANSIBLE_SSH_CONTROL_PATH       = get_config(p, 'ssh_connection', 'control_path', 'ANSIBLE_SSH_CONTROL_PATH', u"%(directory)s/ansible-ssh-%%h-%%p-%%r")
 ANSIBLE_SSH_CONTROL_PATH_DIR   = get_config(p, 'ssh_connection', 'control_path_dir', 'ANSIBLE_SSH_CONTROL_PATH_DIR', u'~/.ansible/cp')
+ANSIBLE_SSH_CONTROL_PATH_FIX   = get_config(p, 'ssh_connection', 'control_path_fix', 'ANSIBLE_SSH_CONTROL_PATH_FIX', False, value_type='boolean')
```

Comment:
@willthames the concern in https://github.com/ansible/ansible/issues/11536 was that %C is not supported by all ssh versions. 

I also intend to make the option on by default once we've vetted the feature and have decided on the final config item name.